.github/workflows/build.yaml

```diff
@@ -21,8 +21,7 @@ jobs:
       - name: run gradle check
         run: ./gradlew check
       - name: publish snapshot
-        # final=false here means SNAPSHOT
-        run: ./gradlew publishMavenPublicationToSonatypeRepository -Prelease=true -Pfinal=false
+        run: ./gradlew publishToSonatype
```

Comment:
@trask Regarding the differences between `publishMavenPublicationToSonatypeRepository` and `publishToSonatype` - The former is more specific, if you have different publications configured and you don't want to publish them all at once (which is the case in the Elastic repo) then you can be more specific and set the name of the publication (in this case [maven](https://github.com/open-telemetry/opentelemetry-android/blob/main/buildSrc/src/main/kotlin/otel.publish-conventions.gradle.kts#L35)) to publish, but if you are fine with publishing all the configured publications then you can just run `publishToSonatype` which ends up running all the specific-publication tasks.

It was set to the specific task name mostly due to a copy/paste from what we do at Elastic, however, since there's only one publication configured in this project, then there's no need to be specific, so I've just made the changes 👍